libs/mixer/melody.h

```diff
@@ -7,9 +7,13 @@
 #define SW_TRIANGLE 1
 #define SW_SAWTOOTH 2
 #define SW_SINE 3 // TODO remove it? it takes space
```

Comment:
Drive-by comment about the sine tone - if the space used for the lookup table is a concern, it should be fairly straightforward to replace the table lookup with an approximating function, i.e. https://en.wikipedia.org/wiki/Bhaskara_I%27s_sine_approximation_formula or similar. For sound generation purposes it should not be necessary to be very accurate.

Reply:
The lookup table is 512 bytes, the interpolation would be probably under 100, so if you feel like doing a PR, that would be great!

Reply:
Done, see https://github.com/microsoft/pxt-common-packages/pull/1178